flink-runtime/src/main/java/org/apache/flink/runtime/io/network/bufferprovider/GlobalBufferPool.java

```diff
@@ -63,8 +64,13 @@ public GlobalBufferPool(int numBuffers, int bufferSize) {
 
 		for (int i = 0; i < numBuffers; i++) {
 			try {
-				byte[] buf = new byte[bufferSize];
-				buffers.add(new MemorySegment(buf));
+				// TODO if (config) {
```

Comment:
The direct memory segments are never used. I think you forgot this TODO here. ;)